core/rag/simple_rag.py

```
@@ -102,18 +102,70 @@ def _score(query: str, text: str) -> float:
+    return {"total_chunks": total_chunks, "sources": sources, "index_loaded": _INDEX is not None}
+
+
+def retrieve_context(query: str, max_chunks: int = 3) -> str:
+    """Return top-k relevant chunks concatenated with brief headers."""
+    items = _get_index()
+    if not items:
+        return ""
+
+# core/rag/simple_rag.py
+
+def _score(query: str, text: str) -> float:
```

Flake8 (7.3.0)

[error] redefinition of unused '_score' from line 91

(F811)